src/appleseed.python/bindcurveobject.cpp

```diff
@@ -0,0 +1,148 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2018 Girish Ramesh, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// appleseed.python headers.
+#include "bindentitycontainers.h"
+#include "dict2dict.h"
+
+// appleseed.renderer headers.
+#include "renderer/api/object.h"
+
+// appleseed.foundation headers.
+#include "foundation/platform/python.h"
+#include "foundation/utility/murmurhash.h"
+#include "foundation/utility/searchpaths.h"
+
+// Standard headers.
+#include <cstddef>
+#include <string>
+
+namespace bpy = boost::python;
+using namespace foundation;
+using namespace renderer;
+using namespace std;
+
+// Work around a regression in Visual Studio 2015 Update 3.
+#if defined(_MSC_VER) && _MSC_VER == 1900
+namespace boost
+{
+    template <> CurveObject const volatile* get_pointer<CurveObject const volatile>(CurveObject const volatile* p) { return p; }
+}
+#endif
+
+namespace
+{
+    auto_release_ptr<CurveObject> create_curve_obj(
+        const string&       name,
+        const bpy::dict&    params)
+    {
+        return auto_release_ptr<CurveObject>(
+                        CurveObjectFactory().create(name.c_str(), bpy_dict_to_param_array(params)));
```

Comment:
Indent the second line once (4 spaces) with respect to the first line.

Reply:
Fixed